language/tools/test_generation/src/abstract_state.rs

```diff
@@ -2,36 +2,77 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use std::collections::HashMap;
-use vm::file_format::SignatureToken;
+use vm::file_format::{Kind, SignatureToken};
 
 /// The BorrowState denotes whether a local is `Available` or
 /// has been moved and is `Unavailable`.
-#[derive(Debug, Clone, PartialEq)]
+#[derive(Debug, Clone, Copy, PartialEq)]
 pub enum BorrowState {
     Available,
     Unavailable,
 }
 
+/// This models a value on the stack or in the locals
+#[derive(Debug, Clone, PartialEq)]
+pub struct AbstractValue {
+    /// Represents the type of the value
+    pub token: SignatureToken,
+
+    /// Represents the kind of the value
+    pub kind: Kind,
```

Comment:
Not even to mention Kind vs Type. What next, a type kind? ;-)

Reply:
I think kinds are pretty standard. In any case, this is what is actually used in Move so I can't avoid using it.